solidity/contracts/hooks/OptimismMessageHook.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {IOptimismMessageHook} from "../interfaces/hooks/IOptimismMessageHook.sol";
+import {OptimismIsm} from "../isms/native/OptimismIsm.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+
+// ============ External Imports ============
+import {ICrossDomainMessenger} from "@eth-optimism/contracts/libraries/bridge/ICrossDomainMessenger.sol";
+
+contract OptimismMessageHook is IOptimismMessageHook {
+    uint32 public constant OPTIMISM_DOMAIN = 10;
```

Comment:
Consider putting this in the initializer. Presumably this hook would be useful for anything built on OP stack? Including optimism testnets

Reply:
Whoops I meant constructor